Command-Line.md

```diff
@@ -583,35 +583,36 @@ Clients must contain the following file:
 
 ### Generating certificate and fingerprint
 
-The main UI application has built-in functionality to generate a self-signed server certificate,
-and fingerprint. In a command line only ([portable](#portable)) environment you will have to create these manually,
-using the OpenSSL command line utility, which is included in a Barrier installation together
-with a Barrier specific OpenSSL configuration file `barrier.conf`. To create them the same
-way as the UI application does, you can follow the following Windows example.
-It uses `openssl.exe` and `barrier.conf` from a Barrier installed in
-`C:\Program Files\Barrier`, generating configuration in `%LocalAppData%\Barrier\SSL`.
-If you have the OpenSSL files in a different location and/or are planning to keep the SSL files in a
-custom location specified with command line argument `--profile-dir`, you must change the paths in the example accordingly.
-
-```
-mkdir "%LocalAppData%\Barrier\SSL\Fingerprints" >NUL 2>&1
-set OPENSSL_CONF=C:\Program Files\Barrier\barrier.conf
+The main UI application has built-in functionality for handling encryption.
+In server mode it will generate a self-signed server certificate and a fingerprint.
+In client mode it will prompt for you to accept the server's fingerprint, and add
+it to your list of trusted servers.
+In a command line only ([portable](#portable)) environment you will have to handle
+this manually. You can use the OpenSSL command line utility which is included in
+a Barrier installation together with a Barrier specific OpenSSL configuration
+file `barrier.conf`. To create them the same way as the UI application does,
+you can follow the following Windows example. It uses `openssl.exe` and `barrier.conf`
+from a Barrier installed in default location `C:\Program Files\Barrier`, generating
+configuration in default location `%LocalAppData%\Barrier\SSL`. If you have the
+OpenSSL files in a different location and/or are planning to keep the SSL files in
+a custom location specified with command line argument `--profile-dir`, you must
+change the paths in the example accordingly.
+
+```
+MKDIR "%LocalAppData%\Barrier\SSL\Fingerprints" >NUL 2>&1
+SET OPENSSL_CONF=C:\Program Files\Barrier\barrier.conf
 SET RANDFILE=%LocalAppData%\Barrier\SSL\.rnd
 "C:\Program Files\Barrier\openssl.exe" req -x509 -nodes -days 365 -subj /CN=Barrier -newkey rsa:2048 -keyout "%LocalAppData%\Barrier\SSL\Barrier.pem" -out "%LocalAppData%\Barrier\SSL\Barrier.pem"
 IF EXIST "%RANDFILE%" DEL "%RANDFILE%"
-"C:\Program Files\Barrier\openssl.exe" x509 -fingerprint -sha1 -noout -in "%LocalAppData%\Barrier\SSL\Barrier.pem" > "%LocalAppData%\Barrier\SSL\Fingerprints\Local.txt"
+FOR /F "tokens=2 delims=^=" %a in ('""C:\Program Files\Barrier\openssl.exe" x509 -fingerprint -sha1 -noout -in "%LocalAppData%\Barrier\SSL\Barrier.pem""') DO ECHO %a > "%LocalAppData%\Barrier\SSL\Fingerprints\Local.txt"
```

Comment:
Why do we need this...? What does it add to the previous description?

Reply:
This is to strip off the `SHA1 Fingerprint=`, to make the Local.txt content identical to what barrier produces, and then I did not have to specify in the description to only copy the hash and not this prefix when adding to TrustedServers.txt. I know, it looks a bit "verbose"..

I'm considering rewriting this section, even more. Perhaps with PowerShell statements, which is probably a bit easier example to read, or maybe instead go back and just mention the plain openssl commands without the other stuff. I haven't decided yet. I have been experimenting with different solutions (thats why the late response). For instance on how I can use `openssl s_client` on the client side to grab the fingerprint from remote server. Will probably take some time (at least days, perhaps some weeks) for me to finish and conclude, so its up to you if you want to merge this one, else I can make it a draft/work-in-progress kind of PR, or even delete it, until next step is ready.